infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -162,6 +163,10 @@
                     + "              If this limit is exceeded "
                     + "              write requests will be rejected [default: 100.0].\n         "
                     + "                                                                          "
+                    + " --stream-names-file=<stream_names_file>                                  "
```

Comment:
This requires too much manual labor and is not robust to an out dated stream file name. 

IMO the easiest and cleanest way of doing this is to send the table names as metadata in the log write request instead of reverse looking uuid->name based on a local file. This adds a negligible overhead to the write request that can be removed later with a proper stream registry that is replicated.

Reply:
I initially tried this too - it's not a trivial change - I was trying to modify the serialize() function to add the tableName, wasn't sure about the compatibility and chickened out. This is simple, it works and is certainly better than logging UUIDs everywhere.

Reply:
"Why is it better than using UUIDs? that point is not clear to me, most of my debugging I almost never need to know the table name."
Most mortals like me cannot mentally translate thousands of 32 byte hexadecimal strings into their parent table names. Can we please stop pretending that UUIDs are more useful than human readable tableNames in log files?

"The problem with this PR is it can produce inconsistent logs within a server and across servers, this will only make it harder to debug. This approach also introduces manual labor that is not necessary and should be automated."
Every location within CorfuServer should be changed to log tableName instead of UUID, if we are logging a mix of UUID and tableName that's a bug and should be fixed. Do you see any location where UUID is logged in spite of tableName please point that out in your review comments, I will fix it right away. Also not sure what you mean by manual labor? The work to map names to UUIDs is likely something done once and reused many times.

"Generating a CSV zfile from the client side, then copying that file to all nodes of the cluster is not really efficient. Imagine that table mapping changes after it is generated, now the server will produce logs for some tables and UUIDs for others, its not consistent."
The alternative you propose sending table names in addition to stream UUIDs is far more inefficient especially when the UUID is quite literally derived from the tableName. We should be using either one, definitely not both. But that's a more bigger change.

"You mentioned that you need this to for some scripts that you wrote, but if you already have that mapping why cant you write a script that remaps the UUIDs to names, if you really need it. I think that's much cleaner than requiring the server to do this mapping because of the aforementioned points."

Because I am not the only person debugging, there are several engineers who want to triage using corfu.9000.log and simply cannot figure out a way to map UUIDs to table names. This is meant to help them, not me. We do have scripts but it slows down debugging if someone has to run a script to analyze the log. Also browsing log files via UI is now possible.

Reply:
> Most mortals like me cannot mentally translate thousands of 32 byte hexadecimal strings into their parent table names. Why are we debating if UUIDs are more useful than tableNames in log files?

This is a red herring, it doesn't introduce new information and it doesn't answer my question. Let me distill my point of view further so that subsequent responses can be more focused and objective. 

First of all, this PR in its current form will break logging. Currently, on master all logging (except for two places) refers to the stream by the UUID. Your change only remaps the logunit server logs. In essence, this means that logs produced from the clients and other servers cannot be correlated with the logunit server logs. Therefore this PR would degrade the current experience of debugging the system. 


Second, you are making a false assumption that a human readable reference would add more information to assist debugging. Why do you think that tracing a string like "TableA" is easier than tracing a string like "39d4" ? Both are just references and a UUID is more succinct than an arbitrary length string which makes it cleaner to log. Furthermore, unlike a UUID, a string is not specified, so logging a string with special characters can make it really hard to read the log file. 

Of course at some point in the loop, a human readable string is required (for example inspecting metrics logs), but my argument is that step is usually minor step in the debugging process and isn't a good enough reason convert all logging statements from UUID to an arbitrary length string, that is, It's not a better trade-off.


Third, there is a bootstrap problem. This PR's "feature" requires clients to generate the mapping beforehand, but clients can open streams and generate names/ids at runtime after connecting to the server. This means that the mapping generated is not accurate and needs to be consistently updated across all nodes. Furthermore, this also requires to the update mapping on all nodes in the cluster at the "right time" otherwise some servers would log UUIDs and some servers would log the table name string. This makes logging inconsistent. 


When making claims like:

> Every location within CorfuServer should be changed to log tableName instead of UUID

and 

> This is meant to help them, not me

please make sure to supplement your points with a technical argument, or a use case. 

I also find it a little ironic that you reject my anecdotal evidence of debugging 100s of bugs, but use yours as a main point.

Reply:
> > Most mortals like me cannot mentally translate thousands of 32 byte hexadecimal strings into their parent table names. Why are we debating if UUIDs are more useful than tableNames in log files?
> 
> This is a red herring, it doesn't introduce new information and it doesn't answer my question. Let me distill my point of view further so that subsequent responses can be more focused and objective.
> 
> First of all, this PR in its current form will break logging. Currently, on master all logging (except for two places) refers to the stream by the UUID. Your change only remaps the logunit server logs. In essence, this means that logs produced from the clients and other servers cannot be correlated with the logunit server logs. Therefore this PR would degrade the current experience of debugging the system.
>

Why are you commenting on an older commit? Please see the latest commit's files before commenting if you desire to be helpful - I have changed Sequencer logs as well. Most importantly this does not "break logging". Logging is meant *for human beings* first, not machines. Logging something that isn't human readable *is broken*! Also what do you mean by "other servers" are you referring to state transfer? If so please share the exact location where a stream UUID is logged without that your comment is not helpful at all.

> Second, you are making a false assumption that a human readable reference would add more information to assist debugging. Why do you think that tracing a string like "TableA" is easier than tracing a string like "39d4" ? Both are just references and a UUID is more succinct than an arbitrary length string which makes it cleaner to log. Furthermore, unlike a UUID, a string is not specified, so logging a string with special characters can make it really hard to read the log file.

Who exactly cares about succinct or variable lengths in logs? No human who looks at _22932286-9b8f-3469-8a49-816320df0f2a_ will know that this is really the _CorfuSystem$RegistryTable_ so being succinct or fixed length means absolutely nothing and adds ZERO value to the log. Also who exactly is adding "special" characters to table name? Can you please share a real concrete example before throwing a non-existent hypothetical?

> 
> Of course at some point in the loop, a human readable string is required (for example inspecting metrics logs), but my argument is that step is usually minor step in the debugging process and isn't a good enough reason convert all logging statements from UUID to an arbitrary length string, that is, It's not a better trade-off.

Adding steps to debugging is wrong and should be avoided whenever possible. Humans make mistakes and when they do we want to be able to get to the root cause in the FASTEST way possible.

> 
> Third, there is a bootstrap problem. This PR's "feature" requires clients to generate the mapping beforehand, but clients can open streams and generate names/ids at runtime after connecting to the server. This means that the mapping generated is not accurate and needs to be consistently updated across all nodes. Furthermore, this also requires to the update mapping on all nodes in the cluster at the "right time" otherwise some servers would log UUIDs and some servers would log the table name string. This makes logging inconsistent.

This is a valid point and the correct solution is to have a RPC to the server whenever a new client stream is opened which the server can store and use for things like logging. We can certainly debate on the best approach for solving this problem for generic arbitrary names but that change can go **in addition** to this current approach which works and is certainly better than status quo.

> 
> When making claims like:
> 
> > Every location within CorfuServer should be changed to log tableName instead of UUID
> 
> and
> 
> > This is meant to help them, not me
> 
> please make sure to supplement your points with a technical argument, or a use case.
> 
> I also find it a little ironic that you reject my anecdotal evidence of debugging 100s of bugs, but use yours as a main point.

My point, call it technical, philosophical or just practical, is **Logs are meant for Humans first**. Your arguments are all about keeping logs "consistent", "fixed-length" or "concise" which I argue are completely pointless if the log is **unreadable** to begin with.